Add a warning about houston needing to be accessible for SCIM (#2774)
astronomer/docs: software/integrate-auth-system.md

Astronomer Software supports integration with the open standard System for Cross-Domain Identity Management (SCIM). Using the SCIM protocol with Astronomer Software allows you to automatically provision and deprovision users and Teams based on templates for access and permissions. It also provides better observability through your identity provider for when users and Teams are created or modified across your organization.

> :::warning

Reviewer: ```suggestion
:::info
```
Warning is reserved for dangerous actions or breaking changes, I think we can safely downgrade this given that it's a consideration instead of a risk.

> SCIM works by the IdP pushing updates to users & teams to Astronomer Software. For this to work, your Astronomer Software platform must be reachable from the internet. If you are running Astronomer Software without exposing it to the internet, there may be solutions for routing SCIM traffic depending on the combination of cloud provider and IdP. Please contact [Astronomer support](https://support.astronomer.io) for more information.

Reviewer 2: ```suggestion
SCIM works because the IdP pushes updates about users and teams to Astronomer Software. This means your Astronomer Software platform must be connected to the internet to receive those updates. If you are running Astronomer Software without exposing it to the internet, there might be solutions for routing SCIM traffic depending on your combination of cloud provider and IdP. Please contact [Astronomer support](https://support.astronomer.io) for more information.
```
